modules/migrations/gitlab.go

```diff
@@ -85,15 +86,14 @@ type GitlabDownloader struct {
 // NewGitlabDownloader creates a gitlab Downloader via gitlab API
 //   Use either a username/password, personal token entered into the username field, or anonymous/public access
 //   Note: Public access only allows very basic access
-func NewGitlabDownloader(baseURL, repoPath, username, password string) *GitlabDownloader {
+func NewGitlabDownloader(baseURL, repoPath, username, password, token string) *GitlabDownloader {
 	var gitlabClient *gitlab.Client
 	var err error
-	if username != "" {
-		if password == "" {
-			gitlabClient, err = gitlab.NewClient(username, gitlab.WithBaseURL(baseURL))
-		} else {
-			gitlabClient, err = gitlab.NewBasicAuthClient(username, password, gitlab.WithBaseURL(baseURL))
-		}
+	gitlabClient, err = gitlab.NewClient(token, gitlab.WithBaseURL(baseURL))
```

Comment:
why bother with the var gitlabClient and var error above when you can just
```suggestion
	gitlabClient, err := gitlab.NewClient(token, gitlab.WithBaseURL(baseURL))
```

Reply:
this is true ... and should be changed upstream too then ...

Reply:
this should have been done in #12862 :D

Reply:
it's not a problem... it can stay in here; just something I noted.

Reply:
#13103